### PR TITLE
[versavis] Add option to apply CLAHE (histogram equalizat4ion) in the driver

### PR DESCRIPTION
**Pull request by mfehr**
Since we use CLAHE in many places now it seems having the option to do it directly in the ROS driver seems a practical solution. I added the option to work on BGR, RGB and MONO, with this we can actually have histogram equalized color images that could help with the detector in dark environments.

### CODE REVIEW

**Comment by marco-tranzatto** [comment]
@timonh please test it on the robot, see how it goes and report back to @mfehr

File: spinnaker_camera_driver/launch/camera.launch [hunk 1]
```diff
@@ -60,6 +60,9 @@ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSI
       <!-- Use the camera_calibration package to create this file -->
       <param name="camera_info_url" if="$(arg calibrated)"
              value="file://$(env HOME)/.ros/camera_info/$(arg camera_serial).yaml" />
+      <param name="histogram_equalization_enabled" value="true" />
```
Comment (marco-tranzatto) [inline spinnaker_camera_driver/launch/camera.launch]: do you want the default value in the example launch file set to true?

Comment (mfehr) [reply spinnaker_camera_driver/launch/camera.launch]: fixed

File: spinnaker_camera_driver/src/nodelet.cpp [hunk 2]
```diff
@@ -621,6 +636,81 @@ class SpinnakerCameraNodelet : public nodelet::Nodelet {
...
+                  cv::Mat ycrcb;
+                  cv::cvtColor(image_to_process, ycrcb, CV_RGB2YCrCb);
+                  std::vector<cv::Mat> channels;
+                  cv::split(ycrcb, channels);
+
+                  static cv::Ptr<cv::CLAHE> clahe = cv::createCLAHE(
+                      histogram_equalization_clahe_clip_limit,
+                      cv::Size(histogram_equalization_clahe_grid_size,
+                               histogram_equalization_clahe_grid_size));
+                  clahe->apply(channels[0], channels[0]);
```
Comment (floriantschopp) [inline spinnaker_camera_driver/src/nodelet.cpp]: One could think of one more option/flag to include standard histogram equalization
```
cv::equalizeHist( src, dst );
```
as I had better results/less artifacts with it in really dark environments.

**Review by floriantschopp — APPROVED** [review]
We could also include standard histogram equalization as an option in really dark environments (left original, middle std hist eqal, right clahe):
2020-01-09-094743_1916x399_scrot

**Pull request merged**